src/tools/opt-dist/src/utils/mod.rs

```diff
@@ -70,6 +70,16 @@ pub fn with_log_group<F: FnOnce() -> R, R>(group: &str, func: F) -> R {
     }
 }
 
+#[allow(unused)]
+pub fn retry_action<F: Fn() -> anyhow::Result<R>, R>(action: F, count: u64) -> anyhow::Result<R> {
+    for _ in 0..count {
+        if let Ok(result) = action() {
+            return Ok(result);
+        }
+    }
```

Comment:
I would suggest printing the error, since otherwise it looks like it might be lost?

I might also suggest adding a sleep between attempts, in case it is a short-lived transient issue.

Reply:
Good point! Changed both.